backend/controllers/post.js

```diff
@@ -9,4 +9,21 @@ module.exports = class Post {
       res.status(500).json({ error: error });
     }
   }
+  static async apiSearchAllpost(req, res, next) {
```

Comment:
move your this function to services, all your changes

Reply:
create a function in services, and call the function in controller

Reply:
read comments and make changes

Reply:
And also for your next changes and branch make sure to go checkout main and pull it then create a new branch from there